packages/runtime/container-runtime/src/gc/garbageCollection.ts

```diff
@@ -374,12 +376,18 @@ export class GarbageCollector implements IGarbageCollector {
 			return;
 		}
 
-		// Initialize the deleted nodes from the snapshot. This is done irrespective of whether sweep is enabled or not
+		// Initialize the deleted nodes from the snapshot. This is done irrespective of whether GC / sweep is enabled
 		// to identify deleted nodes' usage.
 		if (baseSnapshotData.deletedNodes !== undefined) {
 			this.deletedNodes = new Set(baseSnapshotData.deletedNodes);
 		}
 
+		// If GC is disabled, the GC and tombstone state shouldn't be tracked because there shouldn't be any logs or
+		// errors from GC. The exception is deleted node usage which is mentioned above.
+		if (!this.shouldRunGC) {
```

Comment:
...I wrote all the below basically arguing we should get rid of `shouldRunGC`, and then realized we have to have that for that case where code with old GC Version opens a document with more recent GC Version written. So... yeah we should keep the test option to be able to test this codepath.  I still maintain that the less these codepaths diverge the better, since it's such a rare situation


> Thinking out loud - Where I previously argued that we should keep support for a test config to disable running GC, as an "escape valve", here we see the downside - there are unique codepaths that will never be hit in production until some hypothetical emergency. How do we know it'll fix whatever problem is going on? Seems kind of weak in retrospect.
> 
> In other words - Is it worth the cost of test coverage for these branches, and do we even trust that test coverage?  I'm thinking of the recent example where GC's config API allowed you to disable GC for blobs or datastores independent of each other, but there was a significant bug if you did so. We missed the bug in test coverage when it was introduced, obviously.
> 
> If having the "disable GC" option was "free", then yeah why not have it. But it's not, and I can't think of a motivation (besides "FUD" - Fear Uncertainty & Doubt) to have it.  We can disable Sweep and/or Tombstone enforcement, but we should not support two different codepaths for whether GC data is even generated or not.

Reply:
In that case, I actually think re-running GC with the old version should be just fine. Yes, there may be some trashing but its such a rare scenario that it should not be expensive.

The system also becomes simpler - Every client runs GC with the version that it knows and starts fresh if it recognizes a different version.

Reply:
Also, the only reason to regenerate state if GC is disabled is to clear the unreferenced flag in summary. That code path will also go away soon since the only place its used is for search for which a replacement is being deployed.